lib/dataTrigger/DataTrigger.js

```diff
@@ -0,0 +1,61 @@
+const DataTriggerExecutionResult = require('./DataTriggerExecutionResult');
+const DataTriggerExecutionError = require('../errors/DataTriggerExecutionError');
+
+class DataTrigger {
+  /**
+   *
+   * @param {string} documentType
+   * @param {number} documentAction
+   * @param {function(Document, DataTriggerExecutionContext):DataTriggerExecutionResult} trigger
+   */
+  constructor(documentType, documentAction, trigger) {
```

Comment:
I think we shouldn't pass `documentType` and `documentAction` separately from the contract.